Update to 1.4.4

Pop000100 opened the pull request:
- Update Localization format(repurposed settings and new strings so need your touch)
- Rename and reparameter 1.4.4 functions as necessary
- Add Back command and button(Working now)
+ Support 8 different difficulty settings for (For The worthy and other possible seed support down the line)
+ Seperate difficulty settings
+ Add NPC Ai setting

Pop000100 committed:
Update to 1.4.4

- Update Localization format(repurposed settings and new strings so need your touch)
+ Support 8 different difficulty settings for (For The worthy and other possible seed support down the line)
- Rename and reparameter 1.4.4 functions as necessary
- Add Back command and button(Working now)
+ Seperate difficulty settings
+ Add NPC Ai setting

gardenappl commented on Localization/ru-RU.hjson:
```diff
@@ -1,65 +1,105 @@
-				'''
-		}
-
+		// ItemName.FakeDemonHeart: Fake Demon Heart
+		/* ItemTooltip.FakeDemonHeart:
+			'''
+			You're not supposed to have this item in your inventory!
+			Drop it on the ground to get a regular Demon Heart back.
+			''' */
```
```suggestion
			ItemName.FakeDemonHeart: Фейковое сердце демона
			ItemTooltip.FakeDemonHeart:
			'''
			Этот предмет не должен находиться у вас в инвентаре!
			Выкиньте его, чтобы получить нормальное Сердце.
			'''
```

Pop000100 replied:
This is leftovers from the conversion process.
The converter tends to make copies then point the correct strings to the originals. so I went through and put the strings directly in the spots, to cut down on size... seems I left it in.

gardenappl replied:
```suggestion
```

gardenappl commented on Localization/ru-RU.hjson:
```diff
@@ -1,65 +1,105 @@
+			''' */
 		NowNormalMode: Этот мир теперь в Нормальном режиме.
 		NowExpertMode: Этот мир теперь в Режиме эксперта!
+		// NowMasterMode: The world is now in Master Mode!
```
```suggestion
		NowNormalMode: Этот мир теперь в классическом режиме.
		NowExpertMode: Этот мир теперь в режиме эксперта!
		NowMasterMode: The world is now in режиме мастера!
```

gardenappl commented on Localization/ru-RU.hjson:
```diff
@@ -1,65 +1,105 @@
+		// NowMasterMode: The world is now in Master Mode!
 		AlreadyNormalMode: Этот мир уже в Нормальном режиме.
 		AlreadyExpertMode: Этот мир уже в Режиме эксперта!
+		// AlreadyMasterMode: The world is already in Master Mode!
```
```suggestion
		AlreadyNormalMode: Этот мир уже в классическом режиме.
		AlreadyExpertMode: Этот мир уже в режиме эксперта!
		AlreadyMasterMode: Этот мир уже в режиме мастера!
```

gardenappl commented on Localization/ru-RU.hjson:
```diff
@@ -1,65 +1,105 @@
 		AlreadyNormalMode: Этот мир уже в Нормальном режиме.
 		AlreadyExpertMode: Этот мир уже в Режиме эксперта!
+		// AlreadyMasterMode: The world is already in Master Mode!
 		ToggleModePermission: Включать/выключать Режим эксперта
```
```suggestion
		ToggleModePermission: Включать/выключать режим эксперта
```

gardenappl commented on Localization/ru-RU.hjson:
```diff
@@ -1,65 +1,105 @@
+		// AlreadyMasterMode: The world is already in Master Mode!
 		ToggleModePermission: Включать/выключать Режим эксперта
 		SwitchToNormal: Перейти в Нормальный режим
 		SwitchToExpert: Перейти в Режим эксперта
```
```suggestion
		SwitchToNormal: Перейти в классический режим
		SwitchToExpert: Перейти в режим эксперта
```

gardenappl commented on Localization/ru-RU.hjson:
```diff
@@ -1,65 +1,105 @@
-			$parentVal: Настройки
-			ServerBlocked: Только владелец сервера может менять эти настройки.
+		// SwitchToMaster: Switch to Master Mode
+		// DifficultyCommandUsage: Usage: /difficulty <0(Normal)|1(Expert)|2(Master)>
```
```suggestion
		SwitchToMaster: Перейти в режим мастера
		DifficultyCommandUsage: Использование: /difficulty <0 (классический)|1 (эксперт)|2 (мастер)>
```

gardenappl commented on Localization/ru-RU.hjson:
```diff
@@ -1,65 +1,105 @@
+					Выкиньте его, чтобы получить нормальное Сердце.
 					'''
 			}
+		}
```
Wait so which Demon Heart string is correct? This one or the one above?

Pop000100 replied:
This is the correct one.

gardenappl commented on Localization/en-US.hjson:
```diff
@@ -22,56 +16,86 @@ Mods: {
-				Desc: Add a button that toggles difficulty mode to the Cheat Sheet mod and HERO's Mod.
-			}
+				CurrentFakedDifficulty: {
+					Label: Current Faked Difficulty
+					Tooltip: Makes the enabled settings think it is set to this difficulty.
```
```suggestion
					Label: Fake Difficulty
					Tooltip: Extra challenging features (e.g. boss AI) will function like on this difficulty.
```

Pop000100 replied:
Sounds good to me. writing this stuff isn't my strong suit.

gardenappl commented on Localization/en-US.hjson:
```diff
@@ -22,56 +16,86 @@ Mods: {
+
+				FakedBeneficialDifficulty: {
+					Label: Faked Beneficial Difficulty
+					Tooltip: ""
```
```suggestion
					Tooltip: Beneficial features (e.g. boss drops) will function as if you're playing on this difficulty.
```

Pop000100 replied:
I thought I gave a tooltip to this. looks better then the one I wrote.

gardenappl commented on Localization/en-US.hjson:
```diff
@@ -22,56 +16,86 @@ Mods: {
+					Tooltip: ""
+				}
+
+				ForTheWorthy: {
+					Label: For The Worthy
+					Tooltip: ""
+				}
```
```suggestion
				/* ForTheWorthy: {
					Label: For The Worthy
					Tooltip: ""
				} */
```

Pop000100 replied:
Do we need to comment this out it enlarges the file by X bytes but it doesn't hurt since it is unused since I commented out the code. I'll will let you decide it isn't really important.

gardenappl replied:
Well I saw the localization string and then went searching for where it's used. Took me a while to realize it's unused. So I decided to make this more explicit.

gardenappl commented on Localization/en-US.hjson:
```diff
@@ -22,56 +16,86 @@ Mods: {
+					Tooltip:
+						'''
+						Bring Expert Mode Enemy AI into Normal Mode.
+						(Warning: may break some modded bosses)
```
```suggestion
						Non-boss enemy AI will change according to the selected Fake Difficulty.
						(Warning: may break some modded enemies)
```

gardenappl commented on Localization/en-US.hjson:
```diff
@@ -22,56 +16,86 @@ Mods: {
+					Label: Change Boss AI
+					Tooltip:
+						'''
+						Bring Expert Mode boss AI into Normal Mode.
```
```suggestion
						Boss AI will change according to the selected Fake Difficulty.
```

gardenappl commented on Localization/en-US.hjson:
```diff
@@ -22,56 +16,86 @@ Mods: {
-			}
+				DropTreasureBagsInNormal: {
+					Label: Modify loot
+					Tooltip: Changes loot in accordence with your selected Beneficial Difficulty.
```
```suggestion
					Tooltip: Changes boss loot according to your selected Beneficial Difficulty.
```

gardenappl commented on Localization/en-US.hjson:
```diff
@@ -22,56 +16,86 @@ Mods: {
+			FakeDemonHeart: {
+				DisplayName: Fake Demon Heart
+				Tooltip: ""
+			}
```
Again I can't tell which localization key is correct, there is another one above in the same file.

Pop000100 replied:
The one in BossExpertise.Items.FakeDemonHeart not BossExpertise.ItemName.FakeDemonHeart. Which is empty so we need to move it into the correct location. It is simpler if you do the localization stuff so it stays mergeable since if i want to make changes to it now I have include your changes to solve merge conflicts.

gardenappl replied:
Alright, committing lang changes

gardenappl commented on BossExpertiseSystem.cs:
```diff
@@ -6,36 +6,44 @@ namespace BossExpertise
     class BossExpertiseSystem : ModSystem
 	{
-        public override void OnWorldLoad()
+		static public Difficulty worldDifficulty = new();
```
I propose renaming this to `OriginalDifficulty`.

Pop000100 replied:
Ok, but I will have to modify that when I allow the command to modify the world. So that the before and afters can correctly set the new level.

So think if that jives with your understanding of original.

gardenappl replied:
Mm yeah true. On second thought, WorldDifficulty is fine.

gardenappl commented on Localization/ru-RU.hjson:
```diff
@@ -1,65 +1,105 @@
-				Desc: Добавляет админ-команду /expert, переключающую Режим эксперта.
-			}
+				CurrentFakedDifficulty: {
+					Label: Current Faked Difficulty
+					Tooltip: Makes the enabled settings think it is set to this difficulty.
```
```suggestion
					Label: Фейковая сложность
					Tooltip: Геймплейные фичи, опасные для игрока (например, поведение боссов) будут вести себя как на этой сложности.
```

gardenappl commented on Localization/ru-RU.hjson:
```diff
@@ -1,65 +1,105 @@
+
+				AddExpertCommand: {
+					Label: Команда /difficulty
+					Tooltip: Добавляет админ-команду /difficulty, переключающую Режим эксперта.
```
```suggestion
					Tooltip: Добавляет админ-команду /difficulty, переключающую сложность мира.
```

gardenappl commented on Localization/ru-RU.hjson:
```diff
@@ -1,65 +1,105 @@
+				}
+
+				FakedBeneficialDifficulty: {
+					// Label: Faked Beneficial Difficulty
+					// Tooltip: ""
```
```suggestion
					Label: Полезная фейковая сложность
					Tooltip: Геймплейные фичи, полезные для игрока (например, лут из боссов) будут вести себя как на этой сложности.
```

gardenappl commented on Localization/ru-RU.hjson:
```diff
@@ -1,65 +1,105 @@
+				}
+
+				ChangeNPCAI: {
+					Label: Change NPC AI
+					Tooltip:
+						'''
+						Bring Expert Mode NPC AI into Normal Mode.
+						(Warning: may break some modded NPCs)
```
```suggestion
					Label: Менять ИИ врагов
					Tooltip:
						'''
						Изменить поведение врагов (за исключением боссов) в зависимости от выбраной "фейковой сложности.
						(Может не работать с врагами из других модов.)
```

gardenappl commented on Localization/ru-RU.hjson:
```diff
@@ -1,65 +1,105 @@
+				}
+
+				BossWhitelist: {
+					// Label: Boss Whitelist
+					/* Tooltip:
+						'''
+						These NPCs will be affected by Boss Expertise
+						even if they're technically not bosses.
+						''' */
```
```suggestion
					Label: Белый список боссов
					Tooltip:
						'''
						Boss Expertise будет влиять на этих врагов так, будто они боссы
						(даже если технически не являються боссами)
						'''
```

gardenappl commented on Localization/ru-RU.hjson:
```diff
@@ -1,65 +1,105 @@
+				}
+
+				SlotsWorksInNormal: {
+					// Label: Accessory Slots
+					// Tooltip: The extra accessory slot(s) work based on the current beneficial difficulty.
```
```suggestion
					Label: Слоты аксессуаров
					Tooltip: Дополнительные слоты аксессуаров изменяться в зависимости от "полезной фейковой сложности".
```

gardenappl commented on Localization/ru-RU.hjson:
```diff
@@ -1,65 +1,105 @@
-					даже если они технически не являются боссами.
-					'''
+				ChangeBossAI: {
+					Label: Change Boss AI
+					Tooltip:
+						'''
+						Bring Expert Mode boss AI into Normal Mode.
+						(Warning: may break some modded bosses)
+						'''
```
```suggestion
					Label: Менять ИИ боссов
					Tooltip:
						'''
						Изменить поведение боссов в зависимости от выбраной "фейковой сложности.
						(Может не работать с боссами из других модов.)
						'''
```

gardenappl commented on Localization/ru-RU.hjson:
```diff
@@ -1,65 +1,105 @@
 
-			DemonHeartHack: Сердце демона работает в Нормальном режиме
+				DropTreasureBagsInNormal: {
+					// Label: Modify loot
+					// Tooltip: Changes loot in accordence with your selected Beneficial Difficulty.
```
```suggestion
					Label: Менять лут
					Tooltip: Меняет дропы из боссов в зависимости от "полезной фейковой сложности".
```

gardenappl reviewed (commented):
So I updated the ru-RU strings (will commit them shortly), but I also left some suggestions for en-EN strings (some are admittedly nitpicks) and some questions.

gardenappl commented on Localization/en-US.hjson:
```diff
@@ -22,56 +16,86 @@ Mods: {
-					'''
+				AddExpertCommand: {
+					Label: Add /difficulty command
+					Tooltip: Add the /difficulty chat command that toggles Expert Mode.
```
```suggestion
					Tooltip: Add the /difficulty chat command that changes the world's actual difficulty.
```

Pop000100 replied:
the command changes the faked difficulty. I would need more reflection to modify the variable since last time I checked it is private/internal.

I don't think it ever targeted the World's Setting.

Edit: It doesn't seem that hard. So if it is desired ill change it if not, I found a bug so that will need to be fixed either way.

gardenappl replied:
Oh sorry I was mistaken. But I think originally that's basically what I wanted from the command. The fake difficulty being managed by the config and the actual difficulty being managed by the command.

gardenappl replied:
Btw there are a bunch of "resolved conversations" from my ru-RU lang commit, but there are also some unresolved convos that GitHub decided to hide for whatever reason.

gardenappl committed:
Additional ru-RU localization

gardenappl commented on Localization/en-US.hjson:
```diff
@@ -1,17 +1,11 @@
-				'''
-		}
-
+		ItemName.FakeDemonHeart: Fake Demon Heart
+		ItemTooltip.FakeDemonHeart:
+			'''
+			You're not supposed to have this item in your inventory!
+			Drop it on the ground to get a regular Demon Heart back.
+			'''
```
```suggestion
```

gardenappl reviewed (commented):
Just removing unused lang stuff

gardenappl committed:
Localization cleanup

Pop000100 commented:
I got busy haven't had to time to setup a second computer, to test the next commit. Will probably be committed the 7th, 8th, or 9th

Pop000100 committed:
Make the command target the world

- Make the command target the world
- Fix the multiplayer sync(this was broken for while). None complained it might not have been needed.
- remove someuneeded logging.
- change a critical if to an elseif line 32 of expert command(the mentioned culprit for the bug that I said wait for.)

gardenappl commented:
Thanks a lot!
